ethers-ffi/src/lib.rs

```diff
@@ -264,27 +279,19 @@ pub mod android {
 
     #[no_mangle]
     pub extern "system" fn Java_com_uniswap_EthersRs_generateMnemonic(env: JNIEnv, _class: JClass) -> jobject{
-        let mnemonic_struct = generate_mnemonic();
+        let mut mnemonic_struct = generate_mnemonic_rust();
 
         let class_name = "com/uniswap/CMnemonicAndAddress";
         let class = env
             .find_class(class_name)
             .expect(&format!("Failed to find class: {}", class_name));
 
-        // Convert the mnemonic and address fields to Rust String
-        let mnemonic = unsafe { CStr::from_ptr(mnemonic_struct.mnemonic).to_string_lossy().into_owned() };
-        let address = unsafe { CStr::from_ptr(mnemonic_struct.address).to_string_lossy().into_owned() };
-
         // Create a new Java string from the Rust string
-        let mnemonic_jstring = rust_string_to_jstring(&env, mnemonic);
-        let address_jstring = rust_string_to_jstring(&env, address);
+        let mnemonic_jstring = rust_string_to_jstring(&env, mnemonic_struct.mnemonic.clone());
+        let address_jstring = rust_string_to_jstring(&env, mnemonic_struct.address.clone());
```

Comment:
not sure if these need to be cloned?

Reply:
it doesn't appear to me that we need to clone